extensions/vscode/src/extension.ts

```diff
@@ -19,12 +27,30 @@ import { BlocCodeActionProvider } from "./code-actions";
 
 const DART_MODE = { language: "dart", scheme: "file" };
 
+/// The name of the custom when clause context that is set to true if a Bloc
+/// project is loaded in the workspace, or false otherwise.
+const anyBlocProjectLoadedClause = "bloc:anyBlocProjectLoaded";
+
 export function activate(_context: ExtensionContext) {
   if (workspace.getConfiguration("bloc").get<boolean>("checkForUpdates")) {
     analyzeDependencies();
   }
 
+  updateAnyBlocProjectLoaded();
+
   _context.subscriptions.push(
+    window.onDidChangeActiveTextEditor(updateAnyBlocProjectLoaded),
+    workspace.onDidChangeWorkspaceFolders(updateAnyBlocProjectLoaded),
+    workspace.onDidChangeTextDocument(async (event) => {
```

Comment:
why is `onDidChangeTextDocument` needed? That seems like it would be quite heavy if we're recursively analyzing `pubspec.yaml` files every time a text document changes.